Updating HACKING doc with more detailed information about Kubernetes rebases

fabianofranz opened the pull request.

fabianofranz commented:
@smarterclayton @mfojtik @soltysh take a look if you have anything to add, tks!

smarterclayton commented on HACKING.md:
```diff
@@ -141,17 +141,82 @@ appropriately.  To install `godep` locally run:
-## Updating Godeps from upstream
+## Updating Kubernetes from upstream
+
+There are a few steps involved in rebasing Origin to a new version of Kubernetes. We need to make sure
```
Why doesn't this describe using hack/rebase-kube?

fabianofranz replied:
Good catch, I did it manually last time so forgot about the script when writing this. I'll add it to the doc.

smarterclayton replied:
I'm ok with the explanation, but it should be "use script, do these things, now i'll explain how the script works" probably.

soltysh replied:
Besides the script @smarterclayton mentioned before rebasing k8s is invoking `godep restore` inside origin (https://github.com/openshift/origin/blob/master/hack/rebase-kube.sh#L30-L32). This is needed, because origin has some dependencies which do not exist in k8s, eg. STI (https://github.com/openshift/origin/blob/master/Godeps/Godeps.json#L504-L506). Your below description does not cover that, that means that invoking `godep save` will erase such deps. It'll be worth mentioning that you should check the diff of `Godeps.json` file if it's not missing any deps and if so check if that was intentional.

smarterclayton replied:
Yeah - maybe the script should show the diff first, and say "hey are you ok with all of these"?  It's easy to skip that step (I have as well).  The UPSTREAM: commits might be resetting those versions as well.

soltysh commented on HACKING.md:
```diff
@@ -141,17 +141,82 @@ appropriately.  To install `godep` locally run:
+`<commit SHA>` is the commit id for the Kubernetes version we are including in our Godeps. It can be 
+found in our `Godeps/Godeps.json` in the declaration of any Kubernetes package.
+
+Eventually during the development cycle we introduce changes to dependencies right in the Origin 
```
I'd propose to put this part under separate header. The first part explains how to do full k8s update, whereas this describes cherry-picking just single commits. Both parts makes sense and look ok, but are disjoint IMHO.

fabianofranz commented:
@smarterclayton @soltysh Thanks for the feedback. Just pushed an update, also addresses comments from my briefing meeting with @jwhonce and @markturansky.

https://github.com/fabianofranz/origin/blob/rebase_kube_docs/HACKING.md#updating-kubernetes-from-upstream

smarterclayton commented:
Doc is thorough and excellent, LGTM [merge]

fabianofranz commented:
Looks like the .md is not compiling.  :)

[merge] again

smarterclayton commented:
[merge]
